src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressByMessageFilterTest.java

```diff
@@ -0,0 +1,165 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2021 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.filters;
+
+import static com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck.MSG_METHOD;
+import static com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck.MSG_VARIABLE;
+
+import org.junit.jupiter.api.Test;
+
+import com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport;
+import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+import com.puppycrawl.tools.checkstyle.TreeWalker;
+import com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
+
+public class SuppressByMessageFilterTest extends AbstractModuleTestSupport {
+
+    @Override
+    protected String getPackageLocation() {
+        return "com/puppycrawl/tools/checkstyle/filters/suppressbymessagefilter";
+    }
+
+    @Test
+    public void testDefault() throws Exception {
+        final DefaultConfiguration config = createModuleConfig(SuppressByMessageFilter.class);
+
+        final String[] expectedViolations = {
+            "22:9: " + getCheckMessage(RequireThisCheck.class, MSG_VARIABLE, "a", ""),
+            "23:9: " + getCheckMessage(RequireThisCheck.class, MSG_VARIABLE, "b", ""),
+            "24:9: " + getCheckMessage(RequireThisCheck.class, MSG_METHOD, "foo", ""),
+            "28:9: " + getCheckMessage(RequireThisCheck.class, MSG_VARIABLE, "c", ""),
+        };
+
+        verifySuppressed(config, getPath("InputSuppressByMessageFilter.java"),
```

Comment:
Why is every test using the same input file when you specify the exact configuration to be used inside the input file?
This makes me believe that each test should be using a different input file otherwise each test will just have the same inputs and the same outputs.

Reply:
Yes, I have created separate input files for all the Unit Test Cases.

Reply:
After creating the input files and configuring them with `checkFields` and `checkMethods`, all test cases are passing but now I am getting coverage issues.

I believe that it is happening because I have not configured them in input files? or maybe I have done a mistake in something else.
So I tried to configure `checkFormat` and `messageFormat` in the config file of all input files but then I was getting an error like:
`com.puppycrawl.tools.checkstyle.api.CheckstyleException: cannot initialize module com.puppycrawl.tools.checkstyle.TreeWalker - TreeWalker is not allowed as a parent of com.puppycrawl.tools.checkstyle.filters.SuppressByMessageFilter Please review 'Parent Module' section for this Check in web documentation if Check is standard.`

Reply:
My input config look something like these for the error:

```
/*
com.puppycrawl.tools.checkstyle.checks.coding.RequireThisCheck
checkFields = true
checkMethods = true
validateOnlyOverlapping = false


SuppressByMessageFilter
checkFormat = null
messageFormat = null


*/
```

Reply:
null must be in brackets ```(null)```